cpp/ql/src/Security/CWE/CWE-190/ArithmeticWithExtremeValues.ql

```diff
@@ -65,16 +50,79 @@ predicate causeEffectCorrespond(string cause, string effect) {
   effect = "underflow"
 }
 
-from Expr origin, Operation op, VariableAccess va, string cause, string effect
-where
-  taintedVarAccess(origin, va, cause) and
-  op.getAnOperand() = va and
-  (
+predicate isSink(DataFlow::Node sink, VariableAccess va, string effect) {
+  exists(Operation op |
+    sink.asExpr() = va and
+    op.getAnOperand() = va
+  |
     missingGuardAgainstUnderflow(op, va) and effect = "underflow"
     or
     missingGuardAgainstOverflow(op, va) and effect = "overflow"
-  ) and
-  causeEffectCorrespond(cause, effect)
```

Comment:
What do we lose by dropping this?

Reply:
That's just used to format the alert message. Its definition is simply:
```ql

predicate causeEffectCorrespond(string cause, string effect) {
  cause = "max value" and
  effect = "overflow"
  or
  cause = "min value" and
  effect = "underflow"
}
```
i.e., if the source is a minimum value, we say that the operation may underflow. And if the source is a maximum value, we say that the operation may overflow. This conjunct is added in the `select` at the very bottom of the query so nothing is actually changed.